Source/Particles/Collision/BinaryCollision.H

```diff
@@ -102,6 +116,40 @@ public:
         auto& species1 = mypc->GetParticleContainerFromName(m_species_names[0]);
         auto& species2 = mypc->GetParticleContainerFromName(m_species_names[1]);
 
+        // In case of particle creation, create the necessary vectors
+        const int n_product_species = m_product_species.size();
+        amrex::Vector<WarpXParticleContainer*> product_species_vector;
+        amrex::Vector<SmartCopy> copy_species1;
+        amrex::Vector<SmartCopy> copy_species2;
+        for (int i = 0; i < n_product_species; i++)
+        {
+            auto& product = mypc->GetParticleContainerFromName(m_product_species[i]);
+            product.defineAllParticleTiles();
+            product_species_vector.push_back(&product);
+            SmartCopyFactory copy_factory_species1(species1, product);
+            SmartCopyFactory copy_factory_species2(species2, product);
+            copy_species1.push_back(copy_factory_species1.getSmartCopy());
+            copy_species2.push_back(copy_factory_species2.getSmartCopy());
+        }
+#ifdef AMREX_USE_GPU
+        amrex::Gpu::DeviceVector<SmartCopy> device_copy_species1(n_product_species);
+        amrex::Gpu::DeviceVector<SmartCopy> device_copy_species2(n_product_species);
+        amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice, copy_species1.begin(),
+                              copy_species1.end(), device_copy_species1.begin());
+        amrex::Gpu::copyAsync(amrex::Gpu::hostToDevice, copy_species2.begin(),
+                              copy_species2.end(), device_copy_species2.begin());
+        amrex::Gpu::streamSynchronize();
+        auto copy_species1_data = device_copy_species1.data();
+        auto copy_species2_data = device_copy_species2.data();
+#else
+        auto copy_species1_data = copy_species1.data();
+        auto copy_species2_data = copy_species2.data();
+#endif
+        if (m_have_product_species){
+            species1.defineAllParticleTiles();
+            species2.defineAllParticleTiles();
+        }
```

Comment:
By the way, just to clarify: when exactly do we need to call this function? Is it only at the beginning of routines that create new particles? (which looks like is currently the case) @atmyers